controllers/tab/workflow/WorkflowTabHandler.inc.php

```diff
@@ -65,6 +180,244 @@ protected function getProductionNotificationOptions($submissionId) {
+			$submissionFile = $submissionFileDao->newDataObjectByGenreId(GENRE_CATEGORY_DOCUMENT);
+			$submissionFile->setSubmissionId($this->getSubmission()->getId());
+			$submissionFile->setSubmissionLocale($this->getSubmission()->getLocale());
+			$submissionFile->setGenreId(GENRE_CATEGORY_DOCUMENT);
+			$submissionFile->setFileStage(SUBMISSION_FILE_PRODUCTION_READY);
+			$submissionFile->setDateUploaded(Core::getCurrentDate());
+			$submissionFile->setDateModified(Core::getCurrentDate());
+			$submissionFile->setUploaderUserId($user->getId());
+			$submissionFile->setFileSize(filesize($tempFileName));
+			$submissionFile->setFileType('text/xml');
+			$submissionFile->setOriginalFileName($originalFileName);
+			$submissionFile->setName($originalFileName, null);
+			$submissionFileDao->insertObject($submissionFile, $tempFileName, false);
+
+			return DAO::getDataChangedEvent();
+		} else {
+			return new JSONMessage(false);
+		}
+
+	}
+
+	/***
+	 * @param $request
+	 * @return LinkAction
+	 */
+	protected function displaySchedulePublicationLinkAction($request) {
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		$dispatcher = $request->getDispatcher();
+		$schedulePublicationLinkAction = new LinkAction(
+			'schedulePublication',
+			new AjaxModal(
+				$dispatcher->url(
+					$request, ROUTE_COMPONENT, null,
+					'tab.issueEntry.IssueEntryTabHandler',
+					'publicationMetadata', null,
+					array('submissionId' => $this->getSubmission()->getId(), 'stageId' => $this->getStageId())
+				),
+				__('submission.issueEntry.publicationMetadata')
+			),
+			__('editor.article.scheduleForPublication')
+		);
+		return $schedulePublicationLinkAction;
+	}
+
+	/**
+	 *
+	 * @param $args
+	 * @param $request
+	 * @return JSONMessage
+	 */
+	function uploadFile($args, $request) {
+		import('lib.pkp.controllers.tab.workflow.form.FileUploadFormXML');
+
+		$submissionId = $this->getSubmission()->getId();
+		$uploadForm = new FileUploadFormXML($request, $submissionId, $this->getStageId(), $this->getUploaderRoles(), $this->getFileStage());
+
+		$uploadForm->readInputData();
+		if ($uploadForm->validate()) {
+			$uploadedFile = $uploadForm->execute();
+			if (!is_a($uploadedFile, 'SubmissionFile')) {
+			} else {
+
+				$revisionId = 1;
+
+				$templateMgr = TemplateManager::getManager($request);
+				$csrfToken = $request->getSession()->getCSRFToken();
+				$templateMgr->assign(array(
+						'csrfToken' >= $csrfToken,
+						'fileId' >= $uploadedFile->getFileId(),
+						'revision' >= $revisionId,
+						'submissionId' >= $submissionId,
+						'uploaderRoles' >= $this->getUploaderRoles(),
+						'stageId' >= $this->getStageId(),
+					)
+				);
+
+				import('lib.pkp.controllers.tab.workflow.form.FileUploadConfirmationForm');
+				$fileUploadConfirmation = new FileUploadConfirmationForm($request, $uploadedFile->getFileId(), $revisionId, $submissionId, $this->getStageId(), $this->getFileStage(), $request->getSession()->getCSRFToken());
```

Comment:
I think this class is unnecessary, since all it does is format a template file, which you can do directly through PKPTemplateManager (similar to what the normal file upload wizard does to display its confirmation message).